packages/client/src/apis/spot/SpotExecuteAPI.ts

```diff
@@ -47,6 +47,6 @@ export class SpotExecuteAPI extends BaseSpotAPI {
       config.token,
       this.context.signerOrProvider,
     );
-    return erc20.mint(erc20.signer.getAddress(), params.amount);
+    return erc20.mint(await this.getChainSignerAddress(), params.amount);
```

Comment:
@frankfka qq - the chain signer can be updated after initial creation of the client right? that's why `getChainSignerAddress` is async? (otherwise we could make it part of the context but if iirc you can update after creation -- just confirming)

Reply:
makes sense to me! i don't think neither of those get in the way of critical APIs anyways so we should be good.